sidecar/app/routes/start.py

```diff
@@ -123,19 +156,27 @@ def iterfile():
 
 
 @router.post("/ipa-query/{query_id}")
-def start_ipa_test_query(
+def start_ipa_query(
     query_id: str,
     commit_hash: Annotated[str, Form()],
     size: Annotated[int, Form()],
     max_breakdown_key: Annotated[int, Form()],
     max_trigger_value: Annotated[int, Form()],
     per_user_credit_cap: Annotated[int, Form()],
     background_tasks: BackgroundTasks,
+    request: Request,
 ):
     # pylint: disable=too-many-arguments
+    query_manager = request.app.state.QUERY_MANAGER
+    if not query_manager.capacity_available:
+        raise HTTPException(status_code=503, detail="Capacity unavailable")
+
+    settings = get_settings()
     role = settings.role
     if role != role.COORDINATOR:
-        raise Exception(f"Sidecar {role}: Cannot start query without coordinator role.")
+        raise IncorrectRoleError(
+            f"Sidecar {role}: Cannot start query without coordinator role."
```

Comment:
same here, we should clearly state what went wrong - `starting a query on H1 instead of Coordinator`, or something like that